rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYamlVisitor.java

```diff
@@ -346,20 +352,60 @@ private Yaml.Scalar mergeScalar(Yaml.Scalar y1, Yaml.Scalar y2) {
         return !s1.equals(s2) && !acceptTheirs ? y1.withValue(s2) : y1;
     }
 
+    /**
+     * Specialized concatAll function which takes the `insertPlace` property into account.
+     */
+    private <T> ListConcat<T> concatAll(List<T> ls, List<T> t, Function<T, String> getValue) {
+        if (insertMode == null || insertMode == Last || t.isEmpty()) {
+            return new ListConcat<>(ListUtils.concatAll(ls, t), -1,-1);
+        }
+
+        List<T> mutatedEntries = new ArrayList<>();
+        boolean hasInsertedBeforeOrAfterElements = false;
+        int insertIndex = -1, closeIndex = -1;
+        for (int i = 0; i < ls.size(); i++) {
+            T existingEntry = ls.get(i);
+            if (!hasInsertedBeforeOrAfterElements && insertMode == Before && insertProperty.equals(getValue.apply(existingEntry))) {
+                hasInsertedBeforeOrAfterElements = true;
+                mutatedEntries.addAll(t);
+                insertIndex = i;
+                closeIndex = i + t.size() - 1;
+            }
+            mutatedEntries.add(existingEntry);
+            if (!hasInsertedBeforeOrAfterElements && insertMode == After && insertProperty.equals(getValue.apply(existingEntry))) {
```

Comment:
I'm seeing some warnings on these `.equals` calls because `insertProperty` might be null. Can we make the IDE not complain about any new lines of code? Typically press `F2` and `Alt + Enter` until done.